resources/install.rb

```diff
@@ -92,27 +102,78 @@
         options '--force-confdef --force-confold'
         action :install
       end
+    elsif node.platform_family? 'windows'
+
+      service "telegraf_#{new_resource.name}" do
+        service_name 'telegraf'
+        action [:stop]
+        only_if { ::Win32::Service.exists?('telegraf') }
+      end
+
+      file_name = "telegraf-#{new_resource.install_version}_windows_amd64.zip"
+      remote_file "#{Chef::Config[:file_cache_path]}/#{file_name}" do
+        source "#{node['telegraf']['download_urls']['windows']}/#{file_name}"
+        checksum node['telegraf']['shasums']['windows']
+        action :create
+      end
+
+      windows_zipfile ENV['ProgramW6432'] do
+        source "#{Chef::Config[:file_cache_path]}/#{file_name}"
+        not_if { ::File.exist?("#{ENV['ProgramW6432']}\\telegraf\\telegraf.exe") }
+        action :unzip
+      end
+
+      directory "#{ENV['ProgramW6432']}\\telegraf\\telegraf.d" do
+        action :create
+      end
+
+      windows_package 'telegraf' do
+        source "#{ENV['ProgramW6432']}\\telegraf\\telegraf.exe"
+        options "--service install --config-directory \"#{ENV['ProgramW6432']}\\telegraf\\telegraf.d\""
+        # rubocop:enable Metrics/LineLength
+        installer_type :custom
+        action :install
+        only_if { !::Win32::Service.exists?('telegraf') }
+      end
     else
       raise "I do not support your platform: #{node['platform_family']}"
     end
   else
-    raise "#{install_type} is not a valid install type."
+    raise "#{new_resource.install_type} is not a valid install type."
   end
 
-  service "telegraf_#{name}" do
+  service "telegraf_#{new_resource.name}" do
     service_name 'telegraf'
     action [:enable, :start]
   end
 end
 
 action :delete do
-  service "telegraf_#{name}" do
+  service "telegraf_#{new_resource.name}" do
     service_name 'telegraf'
     action [:stop, :disable]
   end
 
-  package 'telegraf' do
-    action :remove
+  if node.platform_family? 'windows'
+    if new_resource.install_type == 'package'
+      chocolatey_package 'telegraf' do
+        action :remove
+      end
+    else
+      win_package 'telegraf' do
+        source "#{ENV['ProgramW6432']}\\telegraf\\telegraf.exe"
+        options '--service uninstall'
+        installer_type :custom
+        action :remove
+      end
+
+      directory "#{ENV['ProgramW6432']}\\telegraf" do
+        action :remove
```

Comment:
This resource action is invalid, it should be `:delete`

https://docs.chef.io/resource_directory.html#actions